indico/core/permissions.py

```diff
@@ -144,7 +147,8 @@ def get_unified_permissions(principal):
     """Convert principal's permissions into a list including read and full access."""
     if principal.full_access:
         return {FULL_ACCESS_PERMISSION}
-    return (set(principal.permissions) + {READ_ACCESS_PERMISSION}) if principal.read_access else principal.permissions
+    return ((set(principal.permissions) + {READ_ACCESS_PERMISSION}) if principal.read_access
```

Comment:
why not sth like this

```python

perms = set(principal.permissions)
if principal.read_access:
    perms.add(READ_ACCESS_PERMISSION)
return perms
```

it would be a bit more readable IMO